src/network_event/server_event.rs

```diff
@@ -290,7 +290,14 @@ fn local_resending_system<T: Event>(
     }
 }
 
+/// Clears queued events.
+///
+/// We clear events while waiting for a connection to ensure clean reconnects.
 fn reset_system<T: Event>(mut event_queue: ResMut<ServerEventQueue<T>>) {
+    let num_cleared = event_queue.len();
+    if num_cleared > 0 {
+        warn!("discarding {num_cleared} queued server events due to a disconnect");
+    }
```

Comment:
Just a personal preference:
```suggestion
    if !event_queue.is_empty() {
        warn!("discarding {} queued server events due to a disconnect", event_queue.len());
    }
```